### PR TITLE
Remove unused deps from extension.

### LINKED ISSUE #110: Improve extension npm install time
fewer dependencies?

### PR DESCRIPTION
**Pull request by samccone**
Fixes #110 

The real slowness here can be primarily blamed on npm3 not on any single dep.. soo until other things are finally resolved this is just the reality that we will have to deal with.

### CODE REVIEW

**Comment by brendankenny** [comment]
:arrow_right::elephant::arrow_left:

**Comment by boopathi** [comment]
The following devDependencies from `extension/` can be removed as well. These will be resolved anyway from the `parent_directory/node_modules`.

+ gulp
+ eslint-config-google
+ and others if I've missed

basically not having duplicates in `lighthouse/node_modules` and `lighthouse/extension/node_modules` explicitly.

**Comment by boopathi** [comment]
And since all of them are devDeps, maybe avoid declaring them in `extension/package.json` and declare everything in `lighthouse/package.json` ? - This can reduce to one time `npm install` from the project root. 

The only problem could be the following and do we require support for this ?

```sh
npm install lighthouse-extension
npm run build
```

**Comment by samccone** [comment]
Hey @boopathi, thanks for the insight! Let me provide you with some of the motivations for the current structure.

So the idea is that someone can work on the extension in isolation from the main code. Think of it as a sub project within the main project. By requiring an isolated set of dependencies and no shared knowledge of each other from the parent or the child this sub project keeps its isolation.

File: extension/package.json [hunk 1]
```diff
@@ -9,9 +9,7 @@
     "build": "gulp build"
   },
   "devDependencies": {
-    "babel-core": "^6.5.2",
```
Comment (paulirish) [inline extension/package.json]: the `babel` task doesnt change at all?

File: extension/bower.json [hunk 2]
```diff
@@ -1,10 +0,0 @@
-{
-  "name": "lighthouse",
-  "private": true,
```
Comment (paulirish) [inline extension/bower.json]: kill `bower i` from the readme?

Comment (paulirish) [inline extension/bower.json]: and remove `bowerrc` and remove the `test` folder?

Comment (samccone) [inline extension/bower.json]: k

File: extension/package.json (hunk 1 above)
Comment (samccone) [inline extension/package.json]: nope! since we are using gulp-babel we are all set

File: extension/bower.json (hunk 2 above)
Comment (samccone) [inline extension/bower.json]: done

File: extension/package.json (hunk 1 above)
Comment (samccone) [inline extension/package.json]: done

**Comment by boopathi** [comment]
@samccone But, for `build` or `watch` in `extension/`, you require files from parent which requires dependencies from parent anyways. So you'd have to run `npm install` on `lighthouse/` before building `lighthouse/extension`

**Pull request merged**